android-components/components/support/remotesettings/src/main/java/mozilla/components/support/remotesettings/RemoteSettingsClient.kt

```diff
@@ -0,0 +1,164 @@
+import mozilla.appservices.remotesettings.RsJsonObject
+import mozilla.components.support.base.log.logger.Logger
+import mozilla.components.support.ktx.util.writeString
+import org.json.JSONArray
+import org.json.JSONObject
+import java.io.File
+import java.io.FileNotFoundException
+import java.io.FileReader
+
+/**
+ * Helper class to download collections from remote settings in app services.
+ *
+ * @property storageRootDirectory The top level app-local storage directory.
+ * @property serverUrl The optional url for the settings server. If not specified, the standard server will be used.
+ * @property bucketName The optional name of the bucket containing the collection on the server.
+ * If not specified, the name of the bucket "main" [or the const if updated] will be used.
+ * @property collectionName The name of the collection for the settings server.
+ */
+class RemoteSettingsClient(
+    private val storageRootDirectory: File,
+    private val serverUrl: String = "https://firefox.settings.services.mozilla.com",
+    private val bucketName: String = "main",
+    private val collectionName: String = "",
+) {
+
+    private val config = RemoteSettingsConfig(
+        serverUrl = serverUrl,
+        bucketName = bucketName,
+        collectionName = collectionName,
+    )
+
+    /**
+     * Fetches all records for the collection from remote servers.
+     */
+    suspend fun fetch(): RemoteSettingsResult = withContext(Dispatchers.IO) {
+        try {
+            val serverRecords = RemoteSettings(config).use {
+                it.getRecords().records
+            }
+            RemoteSettingsResult.Success(serverRecords)
+        } catch (e: RemoteSettingsException) {
+            Logger.error(e.message.toString())
+            RemoteSettingsResult.NetworkFailure(e)
+        }
+    }
+
+    /**
+     * Updates the local storage with [records] data.
+     */
+    fun write(file: File, records: List<RemoteSettingsRecord>) {
+        try {
+            val jsonArray = JSONArray()
+            records.forEach {
+                val recordJson = JSONObject().apply {
+                    put("id", it.id)
+                    put("lastModified", it.lastModified.toString())
+                    put("deleted", it.deleted)
+                    put("attachment", it.attachment)
+                    put("fields", it.fields)
+                }
+                jsonArray.put(recordJson)
+            }
+
+            val jsonObject = JSONObject()
+            jsonObject.put("records", jsonArray)
+            AtomicFile(file).writeString { jsonObject.toString() }
+        } catch (e: Exception) {
+            Logger.error(e.message.toString())
+        }
+    }
+
+    /**
+     * Reads all records for a collection  found in the local storage.
+     **/
+    suspend fun read(): RemoteSettingsResult = withContext(Dispatchers.IO) {
+        val path = "${storageRootDirectory.path}/$serverUrl/$bucketName/$collectionName"
+        val file = File(path)
```

Comment:
Since we've decided to be opinionated about the filepath here, I think these could become class properties that can be shared between functions:

```kotlin
    private val config = RemoteSettingsConfig(
        serverUrl = serverUrl,
        bucketName = bucketName,
        collectionName = collectionName,
    )

    private val path = "${storageRootDirectory.path}/$serverUrl/$bucketName/$collectionName"
    private val file = File(path)
```